packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -107,6 +107,8 @@ const TSErrors = makeErrorTemplates(
     InvalidModifiersOrder: "'%0' modifier must precede '%1' modifier.",
     InvalidTupleMemberLabel:
       "Tuple members must be labeled with a simple identifier.",
+    MissingInterfaceName:
+      "'export default interface' must be followed by an identifier.",
```

Comment:
Now that we changed the generic behaviour for interface parsing we want to have a generic error too. `'interface' declarations must be followed by an identifier.` should be good.

Reply:
I've changed the message.